From 792af73ea90e9689703ab526dd444fdb1c2f8e40 Mon Sep 17 00:00:00 2001
From: Nicolas Cavallari <nicolas.cavallari@green-communications.fr>
Date: Thu, 14 Oct 2021 12:25:46 +0200
Subject: [PATCH] meson.build: add an option to not build man pages.

When cross-compiling, the compiled binaries cannot be run on the build
machine, unless one defines a helper (e.g. qemu) in the meson
configuration, which is most of the time undesirable.

If help2man is installed on the build machine, then the build would fail
because of this, so add an option to disable generating the man pages.

With later versions of meson, it will be possible to automatically
detect this condition.

[Retrieved (and backported) from:
https://gitlab.freedesktop.org/mobile-broadband/libmbim/-/commit/792af73ea90e9689703ab526dd444fdb1c2f8e40]
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 meson.build       | 8 +++++++-
 meson_options.txt | 1 +
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 1c5ee2d..59f1727 100644
--- a/meson.build
+++ b/meson.build
@@ -184,8 +184,8 @@ if enable_gtk_doc
   subdir('docs/reference/libmbim-glib')
 endif
 
-help2man = find_program('help2man', required: false)
-if help2man.found()
+enable_man = get_option('man')
+if enable_man
   subdir('docs/man')
 endif
 
@@ -198,6 +203,7 @@ summary({
 output += '    Documentation:         ' + enable_gtk_doc.to_string() + '\n'
 output += '    bash completion:       ' + enable_bash_completion.to_string() + '\n'
 output += '    gobject introspection: ' + enable_gir.to_string() + '\n\n'
+output += '    man pages:             ' + enable_man.to_string() + '\n\n'
 output += '  System paths\n'
 output += '    prefix:                ' + mbim_prefix + '\n'
 output += '    udev base directory:   ' + mbim_username + '\n\n'
diff --git a/meson_options.txt b/meson_options.txt
index 1305957..bfa33ab 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -7,5 +7,6 @@ option('udevdir', type: 'string', value: '', description: 'where udev base direc
 
 option('introspection', type: 'boolean', value: 'true', description: 'build introspection support')
 option('gtk_doc', type: 'boolean', value: false, description: 'use gtk-doc to build documentation')
+option('man', type: 'boolean', value: 'true', description: 'build man pages using help2man')
 
 option('bash_completion', type: 'boolean', value: true, description: 'install bash completion files')
-- 
GitLab

